### PR TITLE
[FE] Updated profile screen so that it uses matches data for each user

### PR DESCRIPTION
**Pull request by ZhimingZhao**
image

### CODE REVIEW

File: src/frontend/src/app/myprofile/myprofile.component.ts [hunk 1]
```diff
@@ -10,24 +12,46 @@ import { AuthService } from '../services/auth/auth.service';
...
+      losses: 0,
+      problemSolved: 0,
     } as PlayerData;
     const playerStats = authService.getPlayerData();
     this.player.picture = playerStats.picture;
     this.player.name = playerStats.name;
   }
+  ngOnInit(): void {
+    this.profileService.getUserStats().subscribe((res) => {
+      console.log(res);
```
Comment (AntonKanug) [inline src/frontend/src/app/myprofile/myprofile.component.ts:37]:
```suggestion
```

Remove log

File: src/frontend/src/app/myprofile/myprofile.component.ts [hunk 2]
```diff
@@ -10,24 +12,46 @@ import { AuthService } from '../services/auth/auth.service';
...
+          won++;
+        }
+        problemSolved += element.problems.length;
+      });
+      const tempPlayerStats = this.player;
+      tempPlayerStats.wins = won;
+      tempPlayerStats.losses = this.userStats.length - won;
+      tempPlayerStats.problemSolved = problemSolved;
+      this.player = tempPlayerStats;
+      console.log(this.userStats.length);
```
Comment (AntonKanug) [inline src/frontend/src/app/myprofile/myprofile.component.ts:52]:
```suggestion
```

Remove log

File: src/frontend/src/app/services/profile/profile.service.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { AuthService } from '../auth/auth.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ProfileService {
+  url = 'http://localhost:3000';
```
Comment (AntonKanug) [inline src/frontend/src/app/services/profile/profile.service.ts:9]: Use environment.apiUrl, see other services

File: src/frontend/src/app/myprofile/myprofile.component.ts [hunk 4]
```diff
@@ -10,24 +12,46 @@ import { AuthService } from '../services/auth/auth.service';
...
+        if (element.win) {
+          won++;
+        }
+        problemSolved += element.problems.length;
+      });
+      const tempPlayerStats = this.player;
+      tempPlayerStats.wins = won;
+      tempPlayerStats.losses = this.userStats.length - won;
+      tempPlayerStats.problemSolved = problemSolved;
+      this.player = tempPlayerStats;
```
Comment (AntonKanug) [inline src/frontend/src/app/myprofile/myprofile.component.ts:51]: Directly update the player, no need for tempPlayerStats

```suggestion
      this.player.wins = won;
      this.player.losses = this.userStats.length - won;
      this.player.problemSolved = problemSolved;
```

File: src/frontend/src/app/myprofile/myprofile.component.html [hunk 5]
```diff
@@ -14,37 +14,26 @@ <h1 class="player-name">{{ player.name }}</h1>
...
+      *ngFor="let match of userStats; let i = index"
     >
+      <h1
+        class="history-text"
+        [ngClass]="{ green: match.win, red: !match.win }"
+      >
+        {{ match.win ? 'Victory' : 'Defeat&nbsp;' }}
+      </h1>
+      <h1 class="history-text">
+        &nbsp; &nbsp; &nbsp; &nbsp; {{ match.problems }}
```
Comment (AntonKanug) [inline src/frontend/src/app/myprofile/myprofile.component.html:32]: Use padding-left instead

**Review by AntonKanug — CHANGES_REQUESTED** [review]
Only environment.apiUrl comment is necessary

File: src/frontend/src/app/myprofile/myprofile.component.ts (hunk 4 above)
Comment (ZhimingZhao) [reply src/frontend/src/app/myprofile/myprofile.component.ts:50]: Temp is needed

**Review by AntonKanug — APPROVED** [review]

File: src/frontend/src/app/myprofile/myprofile.component.html [hunk 6]
```diff
@@ -14,37 +14,26 @@ <h1 class="player-name">{{ player.name }}</h1>
       <h1 class="stat-text1">Wins: {{ player.wins }}</h1>
       <h1 class="stat-text1">Losses: {{ player.losses }}</h1>
       <h1 class="stat-text1">Problem solved: {{ player.problemSolved }}</h1>
```
Comment (YoussefRizkalla) [inline src/frontend/src/app/myprofile/myprofile.component.html:16]: nit: Problem solved -> Problems solved

**Review by YoussefRizkalla — APPROVED** [review]

**Pull request merged**